tests/e2e/framework/framework.go

```diff
@@ -104,6 +107,27 @@ func (f *Framework) ClientConfig() *rest.Config {
 	return ret
 }
 
+// helmDebugLog prints the debug log of the helm client into a file in the test directory.
+func (f *Framework) helmDebugLog(format string, v ...interface{}) {
+	// check if directory for logs exists and create it if not
+	if _, err := os.Stat(filepath.Dir(TestContext.HelmLogFile)); os.IsNotExist(err) {
+		err := os.MkdirAll(filepath.Dir(TestContext.HelmLogFile), 0755)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	}
```

Comment:
Two things here. If `filepath.Dir(TestContext.HelmLogFile)` is `""` this is going to cause issues. Also, one of the benefits of `os.MkdirAll` is that we don't have to explicitly check for existence.

What about:
```suggestion
	// check if directory for logs exists and create it if not
	if dir := filepath.Dir(TestContext.HelmLogFile); dir != "" {
		err := os.MkdirAll(filepath.Dir(TestContext.HelmLogFile), 0755)
		gomega.Expect(err).NotTo(gomega.HaveOccurred())
	}
```
instead?

Another note. This seems like an expensive operation to perform on EVERY logging call (granted that this may only be debug output), but I would strongly suggest that we instantiate and open the logger once instead of opening it on every call.